extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1480,31 +1506,38 @@ private void checkTaskDuration() throws InterruptedException, ExecutionException
         String taskId = taskEntry.getKey();
         TaskData task = taskEntry.getValue();
 
-        if (task.status.isSuccess()) {
-          // If any task in this group has already completed, stop the rest of the tasks in the group and return.
-          // This will cause us to create a new set of tasks next cycle that will start from the offsets in
-          // metadata store (which will have advanced if we succeeded in publishing and will remain the same if publishing
-          // failed and we need to re-ingest)
-          return Futures.transform(
-              stopTasksInGroup(taskGroup),
-              new Function<Object, Map<Integer, Long>>()
-              {
-                @Nullable
-                @Override
-                public Map<Integer, Long> apply(@Nullable Object input)
+        // task.status can be null if any runNotice is processed before kafkaSupervisor is stopped gracefully.
```

Comment:
Is this comment right? It sounds like it would probably be the other way around (stop gracefully happens before run notices are processed)

Reply:
Good catch. Fixed.